rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -2949,7 +2994,60 @@ error_code sys_fs_get_mount_info(ppu_thread&, vm::ptr<CellFsMountInfo> info, u32
 
 error_code sys_fs_mount(ppu_thread&, vm::cptr<char> dev_name, vm::cptr<char> file_system, vm::cptr<char> path, s32 unk1, s32 prot, s32 unk3, vm::cptr<char> str1, u32 str_len)
 {
-	sys_fs.todo("sys_fs_mount(dev_name=%s, file_system=%s, path=%s, unk1=0x%x, prot=0x%x, unk3=0x%x, str1=%s, str_len=%d)", dev_name, file_system, path, unk1, prot, unk3, str1, str_len);
+	sys_fs.warning("sys_fs_mount(dev_name=%s, file_system=%s, path=%s, unk1=0x%x, prot=0x%x, unk3=0x%x, str1=%s, str_len=%d)", dev_name, file_system, path, unk1, prot, unk3, str1, str_len);
+
+	if (!g_ps3_process_info.has_root_perm())
+	{
+		return CELL_ENOSYS;
+	}
+
+	const auto [path_error, vpath] = translate_to_sv(path);
+
+	if (path_error)
+	{
+		return { path_error, vpath };
+	}
+
+	const auto mp = lv2_fs_object::get_mp(vpath);
+	bool success = true;
+
+	if (mp == &g_mp_sys_dev_hdd1)
+	{
+		success = vfs::mount(vpath, lv2_fs_object::get_vfs(vpath).append("/caches/").append(g_ps3_process_info.get_cellos_appname().substr(0, g_ps3_process_info.get_cellos_appname().find_last_of('.'))), true);
```

Comment:
I think this line is a bit too bloated.
It would be more readable like this imo:

```suggestion
const std::string_view appname = g_ps3_process_info.get_cellos_appname();
success = vfs::mount(vpath, fmt::format("%s/caches/%s", lv2_fs_object::get_vfs(vpath), appname.substr(0, appname.find_last_of('.')), true);
```

Reply:
Wise!